samples/cpp/benchmark_app/main.cpp

```diff
@@ -728,9 +732,14 @@ int main(int argc, char* argv[]) {
                 try {
                     nireq = compiledModel.get_property(ov::optimal_number_of_infer_requests);
                 } catch (const std::exception& ex) {
-                    IE_THROW() << "Every device used with the benchmark_app should "
-                               << "support " << ov::optimal_number_of_infer_requests.name()
-                               << " Failed to query the metric for the " << device_name << " with error:" << ex.what();
+                    OPENVINO_ASSERT(false,
+                                    "Every device used with the benchmark_app should "
+                                    "support ",
+                                    ov::optimal_number_of_infer_requests.name(),
+                                    " Failed to query the metric for the ",
+                                    device_name,
+                                    " with error:",
+                                    ex.what());
```

Comment:
```suggestion
                    throw ov::Exception("Every device used with the benchmark_app should " +
                                        "support " +
                                        ov::optimal_number_of_infer_requests.name() +
                                        " Failed to query the metric for the " +
                                        device_name +
                                        " with error:" +
                                        ex.what());
```

What do you think of using `throw ov::Exception` instead of `OPENVINO_ASSERT(false`?

Reply:
updated